### PR TITLE
mulled biocontainer for vardict-java and htslib

### PR DESCRIPTION
**Pull request by eolaniru**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/vardictjava/main.nf [hunk 1]
```diff
@@ -40,6 +40,7 @@ process VARDICTJAVA {
     | ${convert_to_vcf} \\
         ${args2} \\
     > ${prefix}.vcf
+    bgzip ${prefix}.vcf
```
Comment (nvnieuwk) [inline modules/nf-core/vardictjava/main.nf:43]:
```suggestion
    | bgzip --threads ${task.cpus} ${prefix}.vcf
```

This way you don't first write the unzipped file. Can you also add an `args2` to bgzip?

File: modules/nf-core/vardictjava/main.nf [hunk 2]
```diff
@@ -54,7 +55,7 @@ process VARDICTJAVA {
     def prefix = task.ext.prefix ?: "${meta.id}"
 
     """
-    touch ${prefix}.vcf
+    touch ${prefix}.vcf.gz
```
Comment (nvnieuwk) [inline modules/nf-core/vardictjava/main.nf:58]:
```suggestion
    echo '' | gzip > ${prefix}.vcf.gz
```

**Review by nvnieuwk — COMMENTED** [review]
Some comments :)

**Comment by eolaniru** [comment]
ah, brilliant! thanks @nvnieuwk. Not sure where the args2 will go :)

**Comment by nvnieuwk** [comment]
You can add it after the `bgzip` command. Make sure you create the variable too, similar to `args`

**Comment by nvnieuwk** [comment]
You should also fix the tests since you have a new output format. You can do this with `nf-core modules test --update`

**Review by maxulysse — APPROVED** [review]

**Review by nvnieuwk — APPROVED** [review]
LGTM you can merge when all tests pass :)

**Pull request merged**